File: 1/fixed-capacity-stack.go
package main

import (
	"fmt"
	"os"
)

// push, pop, size, isEmpty
type FixedCapacityStack struct {
	a []string
	n int
}

func (stack *FixedCapacityStack) Push(str string) {
	stack.a = append(stack.a, str)
	stack.n++
}

func (stack *FixedCapacityStack) Pop() string {
	if stack.n == 0 {
		return ""
	}

	e := stack.a[len(stack.a)-1]
	stack.a = stack.a[:len(stack.a)-1]
	stack.n--

	return e
}

func (stack *FixedCapacityStack) Size() int {
	return stack.n
}

func (stack *FixedCapacityStack) IsEmpty() bool {
	return stack.n == 0
}

func main() {
	stack := FixedCapacityStack{
		a: []string{},
		n: 0,
	}

	for i := 1; i < len(os.Args); i++ {
		word := os.Args[i]

		if word != "-" {
			stack.Push(word)
		} else {
			fmt.Println(stack.Pop() + " ")
		}
	}

	fmt.Printf("(%d left on stack)\n", stack.Size())
}

File: 1/fixed-capacity-stack.ts
export class Stack<T> {
  private a: T[];
  private size: number;

  constructor() {
    this.a = [];
    this.size = 0;
  }

  push(item: T): void {
    this.size++;
    this.a.push(item);
  }

  pop(): T | undefined {
    if (this.size === 0) return;
    this.size--;
    return this.a.pop();
  }

  getSize(): number {
    return this.size;
  }

  isEmpty(): boolean {
    return this.size === 0;
  }
}

const main = () => {
  const stack = new Stack();
  const args = process.argv.slice(2);

  for (const arg of args) {
    if (arg !== "-") {
      stack.push(arg);
    } else {
      process.stdout.write(`${stack.pop()} `);
    }
  }
  process.stdout.write("\n");

  console.log(`(${stack.getSize()} left on stack)`);
};

// main();
